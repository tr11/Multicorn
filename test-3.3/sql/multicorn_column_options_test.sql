../../test-2.7/sql/multicorn_column_options_test.sql