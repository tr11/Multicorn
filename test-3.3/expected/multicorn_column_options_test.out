CREATE EXTENSION multicorn;
CREATE server multicorn_srv foreign data wrapper multicorn options (
    wrapper 'multicorn.testfdw.TestForeignDataWrapper'
);
CREATE user mapping for postgres server multicorn_srv options (usermapping 'test');
CREATE foreign table testmulticorn (
    test1 character varying options (prefix 'test'),
    test2 character varying
) server multicorn_srv options (
    option1 'option1'
);
select * from testmulticorn limit 1;
NOTICE:  [('option1', 'option1'), ('usermapping', 'test')]
NOTICE:  [('test1', 'character varying'), ('test2', 'character varying')]
NOTICE:  Column test1 options: {'prefix': 'test'}
NOTICE:  BEGIN
NOTICE:  []
NOTICE:  ['test1', 'test2']
NOTICE:  PRECOMMIT
NOTICE:  COMMIT
   test1   |   test2   
-----------+-----------
 test1 1 0 | test2 2 0
(1 row)

ALTER foreign table testmulticorn alter test1 options (set prefix 'test2');
select * from testmulticorn limit 1;
NOTICE:  [('option1', 'option1'), ('usermapping', 'test')]
NOTICE:  [('test1', 'character varying'), ('test2', 'character varying')]
NOTICE:  Column test1 options: {'prefix': 'test2'}
NOTICE:  BEGIN
NOTICE:  []
NOTICE:  ['test1', 'test2']
NOTICE:  PRECOMMIT
NOTICE:  COMMIT
   test1   |   test2   
-----------+-----------
 test1 1 0 | test2 2 0
(1 row)

ALTER foreign table testmulticorn alter test1 options (drop prefix);
select * from testmulticorn limit 1;
NOTICE:  [('option1', 'option1'), ('usermapping', 'test')]
NOTICE:  [('test1', 'character varying'), ('test2', 'character varying')]
NOTICE:  BEGIN
NOTICE:  []
NOTICE:  ['test1', 'test2']
NOTICE:  PRECOMMIT
NOTICE:  COMMIT
   test1   |   test2   
-----------+-----------
 test1 1 0 | test2 2 0
(1 row)

ALTER foreign table testmulticorn alter test1 options (add prefix 'test3');
select * from testmulticorn limit 1;
NOTICE:  [('option1', 'option1'), ('usermapping', 'test')]
NOTICE:  [('test1', 'character varying'), ('test2', 'character varying')]
NOTICE:  Column test1 options: {'prefix': 'test3'}
NOTICE:  BEGIN
NOTICE:  []
NOTICE:  ['test1', 'test2']
NOTICE:  PRECOMMIT
NOTICE:  COMMIT
   test1   |   test2   
-----------+-----------
 test1 1 0 | test2 2 0
(1 row)

DROP EXTENSION multicorn cascade;
NOTICE:  drop cascades to 3 other objects
DETAIL:  drop cascades to server multicorn_srv
drop cascades to user mapping for postgres
drop cascades to foreign table testmulticorn
